### ISSUE TITLE
Cannot use on Windows

### ISSUE THREAD

**Issue by hippadrone** [ktonon/elm-crypto#1]
Hi, I was trying to use this the other day and got the above error.  Here's a link to ellie:

https://ellie-app.com/ptzzyb7sVa1/0

**Comment by ktonon** [comment]
Thanks for the bug report. My unit tests are passing, but possible something different between the browser and node. I'll investigate in a few days.

**Comment by ktonon** [comment]
@hippadrone I do see the error in ellie. However, I'm not getting this error with either of the following setups:

* running unit tests
* `elm reactor` with a small [demo app](https://github.com/ktonon/elm-crypto/blob/master/demo/Main.elm) ([just added](https://github.com/ktonon/elm-crypto/commit/7c185ba8223568e023e5da43c75ff6dbfc7078a3))

You can test the demo yourself by

* cloning this repo
* `cd demo`
* `elm reactor`
* http://localhost:8000/Main.elm

Then you should see the following

<img width="790" alt="screen shot 2017-09-29 at 8 49 04 pm" src="https://user-images.githubusercontent.com/22655/31040647-adab9990-a557-11e7-8aee-01fdd8e9b5b9.png">

I'm also going to test building the demo with webpack and publishing it as an s3 site, but that's for another day.

I'm renaming the title of this issue to "Cannot use with Ellie".

**Comment by ktonon** [comment]
Bundled with webpack, also works:
http://elm-crypto.betweenconcepts.com.s3-website-us-east-1.amazonaws.com/

I'll look into the ellie issue in time, but this is lower priority. @hippadrone let me know if you are getting the `Division by zero` error anywhere other than ellie.

**Comment by hippadrone** [comment]
I'll hopefully give this a try in the next day or so, I had installed the
package via elm-install (so I wasn't using Ellie initially).  Thank for
looking into this,I'll let you know if I find anything else out.

**Comment by hippadrone** [comment]
So I cloned down the demo, and I still got the same error.  I compared the generated js to your webpacked js and it was interesting.  

(I tried to do the webpack on my end, but I couldn't get it to work.  I've never used it before, so there's definitely a chance for user error) 
 
Anyway, this is an example of what I noticed.  Here's the source in SHA\Constants:
```
initialHashValues : Alg -> WorkingVars
initialHashValues alg =
    case alg of
        SHA224 ->
            WorkingVars
                (W 0xC1059ED8)
                (W 0x367CD507)
                (W 0x3070DD17)
                (W 0xF70E5939)
                (W 0xFFC00B31)
                (W 0x68581511)
                (W 0x64F98FA7)
                (W 0xBEFA4FA4)
```

Here's the js that was generated with elm-reactor (this is incorrect):
```
var _ktonon$elm_crypto$Crypto_SHA_Constants$initialHashValues = function (alg) {
	var _p1 = alg;
	switch (_p1.ctor) {
		case 'SHA224':
			return A8(
				_ktonon$elm_crypto$Crypto_SHA_Types$WorkingVars,
				_ktonon$elm_word$Word$W(-1056596264),
				_ktonon$elm_word$Word$W(914150663),
				_ktonon$elm_word$Word$W(812702999),
				_ktonon$elm_word$Word$W(-150054599),
				_ktonon$elm_word$Word$W(-4191439),
				_ktonon$elm_word$Word$W(1750603025),
				_ktonon$elm_word$Word$W(1694076839),
				_ktonon$elm_word$Word$W(-1090891868));
```

And here's the js that you bundled with webpack (which works):
```
var _ktonon$elm_crypto$Crypto_SHA_Constants$initialHashValues = function (alg) {
	var _p1 = alg;
	switch (_p1.ctor) {
		case 'SHA224':
			return A8(
				_ktonon$elm_crypto$Crypto_SHA_Types$WorkingVars,
				_ktonon$elm_word$Word$W(3238371032),
				_ktonon$elm_word$Word$W(914150663),
				_ktonon$elm_word$Word$W(812702999),
				_ktonon$elm_word$Word$W(4144912697),
				_ktonon$elm_word$Word$W(4290775857),
				_ktonon$elm_word$Word$W(1750603025),
				_ktonon$elm_word$Word$W(1694076839),
				_ktonon$elm_word$Word$W(3204075428));
```

For the numbers that are off, it looks like its prepending FFFF FFFF onto what it should be.  For example:
-1056596264 = FFFF FFFF C105 9ED8

**Comment by ktonon** [comment]
Nice work. I've reproduce the error now. Here is what I found.

| | Node 4 | Node 6 |
|---|---|---|
| Mac | ✔️ | ✔️|
| Windows | ❌ | untested|

On windows pasting `0xC1059ED8` into the elm repl gives `-1056596264`. Also trying to do `Crypto.Hash.sha224 ""` throws the division by zero error.

I think I can get around this by [using strings to initialize the default values](http://package.elm-lang.org/packages/ktonon/elm-word/2.1.0/Word-Hex#toWordArray). For example:

```elm
> import Word exposing (Word(..), Size(..))
> Hex.toWordArray Bit32 "C1059ED8"
Array.fromList [W 3238371032] : Array.Array Word.Word
```

I'll try to get a fix in this week. Thanks for your help debugging this @hippadrone

**Comment by ktonon** [comment]
Looking into this a bit more. Using the elm repl on Windows:

```elm
> 0xC1059ED8
-1056596264 : number

> 3238371032
-1056596264 : number

> (Bitwise.shiftLeftBy 16 0xC105) + 0x9ED8
-1056596264 : Int

> 0xC105 * 2^16 + 0x9ED8
3238371032 : number
```

**Comment by ktonon** [comment]
Turns out the discrepancy between signed and unsigned representations were not the issue. There were two different issues in the `elm-word` package.

Fixed by #2. Published as `1.1.1`
